pylint/checkers/misc.py

```diff
@@ -133,16 +152,41 @@ def process_tokens(self, tokens: list[tokenize.TokenInfo]) -> None:
         if not self.linter.config.notes:
             return
         for token_info in tokens:
-            if token_info.type != tokenize.COMMENT:
-                continue
-            comment_text = token_info.string[1:].lstrip()  # trim '#' and white-spaces
-            if self._fixme_pattern.search("#" + comment_text.lower()):
-                self.add_message(
-                    "fixme",
-                    col_offset=token_info.start[1] + 1,
-                    args=comment_text,
-                    line=token_info.start[0],
-                )
+            if token_info.type == tokenize.COMMENT:
+                comment_text = token_info.string[
+                    1:
+                ].lstrip()  # trim '#' and white-spaces
+                if self._comment_fixme_pattern.search("#" + comment_text.lower()):
+                    self.add_message(
+                        "fixme",
+                        col_offset=token_info.start[1] + 1,
+                        args=comment_text,
+                        line=token_info.start[0],
+                    )
+            elif (
+                self.linter.config.check_fixme_in_docstring
+                and self._is_docstring_comment(token_info)
+            ):
+                docstring_lines = token_info.string.split("\n")
+                for line_no, line in enumerate(docstring_lines):
+                    comment_text = (
+                        line.removeprefix('"""').lstrip().removesuffix('"""')
+                    )  # trim '""""' and whitespace
+                    if self._docstring_fixme_pattern.search(
+                        '"""' + comment_text.lower()
+                    ):
+                        self.add_message(
+                            "fixme",
+                            col_offset=token_info.start[1] + 1,
+                            args=comment_text,
+                            line=token_info.start[0] + line_no,
+                        )
+
+    def _is_docstring_comment(self, token_info: tokenize.TokenInfo) -> bool:
+        return (
+            token_info.type == tokenize.STRING
+            and token_info.line.lstrip().startswith('"""')
```

Comment:
Note that a docstring can also start with `'''`. I'm wondering if this should live in this tokeniser checker as I think it is actually quite hard to recognise docstrings on tokens alone.

Have you considered doing it as a checker for `nodes.Module`, `nodes.ClassDef`, etc? Then you can just check if the regex is in the `.doc` attribute.

Reply:
>Note that a docstring can also start with '''

 totally forgot about this, thanks!

>Have you considered doing it as a checker for nodes.Module, nodes.ClassDef, etc? Then you can just check if the regex is in the .doc attribute.

 I was considering the possibility docstrings could appear outside of `nodes.Module` and `nodes.classDef`, so I figured it's best to use the existing token stream to watch for all occurrences. However, you could argue it's not good python practice (?) in the first place to have docstrings outside modules/classes/methods. Agreed the PR doesn't use the safest heuristic to determine if it's a docstring.

So it looks like we could do
1. update `_is_docstring_comment()` to also check `startswith("'''")`
2. refactor to use `nodes.Module`, `nodes.ClassDef`, `nodes.FunctionDef` and we tighten the scope of docstring fixmes
3. update tokenizer with a new docstring token similar to how we have a token.COMMENT type for a comment fixme. (Haven't looked too deep into this, will probably be higher effort)

Totally down to change it to 2, but would users claim false negatives when they try to create docstring fixme's outside of module/classes/function defs? Perhaps it would help if there were a lint message that recommends against docstrings outside of those places. Do we have something like that already?

Reply:
@Pierre-Sassoulas Opinion? I think trying option 1 for now might be fine, we can always refactor to 2 later on. I just thought I would raise the question to see if it was consciously ignored.

Reply:
Yeah we already use tokenizer for comments. If I understand correctly, tokenizer is for instances where there's not a defined node for the check (eg, a comment can appear "anywhere" in the code, so we check for all tokens for that appearance). I tried to follow that logic with docstrings. Hence I piggy back on the tokenizer to examine for occurrences of `"""` and `'''`.  If we decide we only wish to support docstrings in classes, functions, and methods, then I could use a node visitor on those 3 node types.

It may be easiest at this point to go with Option 1 at this point since it'd be straightforward to make the change in the PR. And as Daniel mentioned, we could refactor to 2 in the future. Thoughts?

Reply:
Docstrings are not supposed to be everywhere in a module, that's useless statements otherwise, so a node approach would work. But if comments requires tokenizer, for consistency of aproachs let's go with 1)